pkg/network/network.go

```diff
@@ -18,195 +18,61 @@ import (
 	"errors"
 	"fmt"
 	"net"
-	"os/user"
-	"strconv"
 	"strings"
 
-	"golang.org/x/sys/unix"
-
 	"github.com/jackpal/gateway"
-	"github.com/sirupsen/logrus"
 	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
 )
 
-// FIXME: Add support for more interfaces. See: https://github.com/nubificus/urunc/issues/13
-// FIXME: Discover the veth endpoint name instead of using default "eth0". See: https://github.com/nubificus/urunc/issues/14
-const DefaultInterface = "eth0"
-const DefaultTap = "tap0_urunc"
-
-var ErrEth0NotFound = errors.New("eth0 device not found")
-var netLog = logrus.WithField("subsystem", "network")
+const (
+	DefaultInterface = "eth0" // FIXME: Discover the veth endpoint name instead of using default "eth0". See: https://github.com/nubificus/urunc/issues/14
+	DefaultTap       = "tapX_urunc"
+)
 
 type UnikernelNetworkInfo struct {
 	TapDevice string
-	EthDevice InterfaceInfo
+	EthDevice Interface
+}
+type Manager interface {
+	NetworkSetup() (*UnikernelNetworkInfo, error)
+}
+
+func NewNetworkManager(networkType string) (Manager, error) {
+	switch networkType {
+	case "static":
+		return &StaticNetwork{}, nil
+	case "dynamic":
+		return &DynamicNetwork{}, nil
+	default:
+		return nil, fmt.Errorf("network manager %s not supported", networkType)
+
+	}
 }
 
-type InterfaceInfo struct {
+type Interface struct {
```

Comment:
Is there a particular reason for the renaming?

Reply:
I opted for the `Interface` name to enhance simplicity. I can revert it, if this poses a potential problem.

Reply:
It certainly does not pose a potential problem. It was just not clear why the change took place.